File: Problem437/Solution.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func pathSum(root *TreeNode, sum int) int {
	if root == nil {
		return 0
	}
	return getCount(root, sum) + pathSum(root.Left, sum) + pathSum(root.Right, sum)
}

func getCount(root *TreeNode, sum int) int {
	if root == nil {
		return 0
	}
	var count int = getCount(root.Left, sum-root.Val) + getCount(root.Right, sum-root.Val)
	if root.Val == sum {
		count++
	}
	return count
}

File: Problem437/Solution.java
import Solution.TreeNode;

public class Solution {

	public int pathSum(TreeNode root, int sum) {
		return root == null ? 0 : getCount(root, sum) + pathSum(root.left, sum) + pathSum(root.right, sum);
	}

	private int getCount(TreeNode node, int sum) {
		return node == null ? 0
				: (node.val == sum ? 1 : 0) + getCount(node.left, sum - node.val)
						+ getCount(node.right, sum - node.val);
	}
}